### ISSUE TITLE
[YCQL] Not supported error should be raised for WHERE clause in CREATE INDEX

### ISSUE THREAD

**Issue by nocaway** [yugabyte/yugabyte-db#5363]
The grammar rule in file "parser_gram.y" has WHERE clause without processing it. For compatibility reasons, the fix of this bug should add a compatibility flag that stops the error reporting when it is turned ON.

Correct code should be similar to the following.
```
IndexStmt:
  CREATE opt_unique INDEX opt_concurrently opt_index_name ON qualified_name
  access_method_clause '(' index_params ')' opt_include_clause OptTableSpace opt_where_clause
  opt_index_options {
    if ($14 &&  FLAGS_raise_index_where_clause_error) {
       // WHERE is not support.
       PARSER_UNSUPPORTED(@1);
    }
    $$ = MAKE_NODE(@1, PTCreateIndex, $2, $5, $7, $10, false, $15, $12);
  }
  | CREATE opt_unique INDEX opt_concurrently IF_P NOT_LA EXISTS opt_index_name ON qualified_name
  access_method_clause '(' index_params ')' opt_include_clause OptTableSpace opt_where_clause
  opt_index_options {
    if ($17 && FLAGS_raise_index_where_clause_error) {
       // WHERE is not support.
       PARSER_UNSUPPORTED(@1);
    }
    $$ = MAKE_NODE(@1, PTCreateIndex, $2, $8, $10, $13, true, $18, $15);
  }
;
```

**Comment by kmuthukk** [comment]
Additional context around this issue (for completeness) where we are syntactically/seemingly supporting partial indexes in YCQL, but ignoring it. [This issue likely originates from the fact that the YCQL parser has its roots in the Postgres parser.]

```
ycqlsh:k> describe k.users;
CREATE TABLE k.users (
    id text PRIMARY KEY,
    name text,
    age int
) WITH default_time_to_live = 0
    AND transactions = {'enabled': 'true'};
ycqlsh:k> create index age_idx ON k.users(age) where age IS NOT NULL;
ycqlsh:k> explain select * from users where age = 10;
 QUERY PLAN
---------------------------------------
 Index Scan using k.age_idx on k.users
   Key Conditions: (age = 10)
ycqlsh:k> explain select * from users where age = NULL;
 QUERY PLAN
---------------------------------------
 Index Scan using k.age_idx on k.users
   Key Conditions: (age = NULL)            <--- this suggests that NULLs are getting indexed also. 
                                           <-- With a true partial index, one would expect this query
                                           <-- to have to do a full scan.
```

**Comment by m-iancu** [comment]
Fixed by e04dd35.
